app/views/comments/_form.html.erb

```diff
@@ -1,104 +1,156 @@
+<!-- comment is false in reply & main comment forms; defined for edit forms -->
+<% comment_id_string = comment ? comment.id.to_s : nil %>
 <!-- get_comment_form_id defined in app/helpers/application_helper.rb: -->
-<% comment_form_id = get_comment_form_id(location, local_assigns[:reply_to], nil) %>
-<div class="comment-form-wrapper card-body" style="background-color:#f8f8f8; border: 1px solid #e7e7e7;padding: 18px;">
-  <form id="comment-form-<%= comment_form_id %>" class="comment-form" data-remote="true" action="/comment/create/<%= @node.nid %><%= "?type=question" if local_assigns[:type]=="question" %>" <% if local_assigns[:aid].blank? %>method="post"<% end %>>
-    <% if current_user && current_user.id == @node.uid %>
-      <span class="pull-right" style="color:#ccc;">
-        <a target="_blank" href="https://publiclab.org/comment-templates">Add a template</a>
-      </span>
+<% comment_form_id = get_comment_form_id(location, local_assigns[:reply_to], comment_id_string) %>
+<div 
+  id="comment-form-wrapper-<%= comment_form_id %>" 
+  class="bg-light card card-body comment-form-wrapper"
+  style="
+    background-color: #f8f8f8; 
+    border: 1px solid #e7e7e7;
+    padding: 18px;
+    <%= "display: none" if location == :edit  %>
+  "
+>
+  <form 
+    id="comment-form-<%= comment_form_id %>" 
+    class="comment-form" 
+    <% if location != :edit %>
+      action="/comment/create/<%= @node.nid %>"
+      data-remote="true" 
```

Comment:
This `data-remote="true"` (along with `comment.js`) seems to play a crucial role in form submission in edit forms vs. the others.

Edit comment submissions refresh the page, while reply & main comment form submissions don't!

Funnily enough, this was something that was stumping me in [this comment here](https://github.com/publiclab/plots2/issues/8810#issuecomment-747867020) way back a the beginning of the internship.

Reply:
Like, if you're just searching in VSCode for places where [submitFormAjax](https://github.com/publiclab/plots2/blob/main/app/assets/javascripts/submitFormAjax.js) and [addComment](https://github.com/publiclab/plots2/blob/main/app/assets/javascripts/submitFormAjax.js) are called, you don't find anything. For the longest time, I thought that these functions weren't actually being called.

But now I think they are! They work based on attaching extra classes to particular comment forms. I think. This intrigues me because a similar thing could be done to edit comment forms so they are submitted with AJAX and don't refresh the page.

Reply:
OK, i think i can help here -- the `data-remote` attribute is used by the obscure `rails-ujs` library -- https://guides.rubyonrails.org/working_with_javascript_in_rails.html#remote-elements

However this is so obscure that I think it's just not worth using this technique; we don't have to remove it but let's add an explanatory comment to help future coders.

Reply:
Ahh, very good to know!